### PR TITLE
misc samples: Replace native_posix w native_sim as default platform (part 1)

### PR DESCRIPTION
**Pull request by aescolar**
For
* samples/modules/chre
* samples/modules/lvgl
* samples/drivers/eeprom
* samples/basic
* samples sensing
* samples display
* samples native_tty
* samples kernel
* samples/subsys/tracing

Switch from native_posix to native_sim:   
* In the docs replace the references to native_posix with native_sim.
* Enable these test in natives_sim, both in the twister filters and by adding overlays.
* Switch the default test platform to native_sim from native_posix.

Note the changes for each samples are trivial; there is just quite a few samples changed in the same PR.

------

Background: during this release native_sim is replacing
native_posix as the main host test/development platform.

Doc preview in: https://builds.zephyrproject.io/zephyr/pr/65070/docs/index.html

### CODE REVIEW

**Review by carlescufi — DISMISSED** [review]
LGTM

**Review by jukkar — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

File: doc/services/tracing/index.rst [hunk 1]
```diff
@@ -191,14 +191,14 @@ Using Tracing
 The sample :zephyr_file:`samples/subsys/tracing` demonstrates tracing with
 different formats and backends.
 
-To get started, the simplest way is to use the CTF format with the ``native_posix``
+To get started, the simplest way is to use the CTF format with the ``native_sim``
```
Comment (kartben) [inline doc/services/tracing/index.rst:194]: I think it also needs updating in the README of the actual sample ("Usage for POSIX Tracing Backend" section, to make sure this really becomes the official recommendation for "the simplest way" of doing things).

**Review by cfriedt — DISMISSED** [review]

File: samples/basic/hash_map/README.rst [hunk 2]
```diff
@@ -16,16 +16,16 @@ This is a simple example that repeatedly:
 Building
 ********
 
-This application can be built on native_posix as follows:
+This application can be built on native_sim as follows:
```
Comment (kartben) [inline samples/basic/hash_map/README.rst:19]: Please ignore if you feel likes it's too much of a cosmetic change and too cumbersome in the context of these already tedious PRs, but would be great to have consistency in making references to native_sim a link, just like other samples touches in this PR do

```suggestion
This application can be built on :ref:`native_sim <native_sim>` as follows:
```

**Review by kartben — CHANGES_REQUESTED** [review]
this is great, thanks! My main request for change would be to make sure to update the subsys/tracing README as it's pretty native_posix heavy still, and it should be taken care of as part of this PR IMO

**Review by kartben — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**